ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -46,30 +48,64 @@ export interface SampleRecordCardProps {
   reloadKey?: string;
   warning?: string;
   groupId?: string;
+
+  record?: RecordType;
+  groups?: Models.GroupType[];
+  destinations?: Models.DestinationType[];
 }
 
 // This is exported so we can manipulate it in tests, but is unused elsewhere.
 export const isConfigUI = grouparooUiEdition() === "config";
 
+const cookies = new Cookies();
+
 const getCachedSampleRecordId = (modelId: string): string => {
-  return globalThis.localStorage?.getItem(
-    `sampleRecord:${grouparooUiEdition()}:${modelId}`
-  );
+  if (isBrowser()) {
+    const localStorageRecordId = globalThis.localStorage?.getItem(
+      `sampleRecord:${grouparooUiEdition()}:${modelId}`
+    );
+    if (localStorageRecordId) {
+      setCachedSampleRecordId(modelId, localStorageRecordId);
+      globalThis.localStorage?.removeItem(
+        `sampleRecord:${grouparooUiEdition()}:${modelId}`
+      );
+      return localStorageRecordId;
+    } else {
+      return cookies.get(`sampleRecord:${grouparooUiEdition()}`)?.[modelId];
+    }
+  }
+  return undefined;
 };
 
 const setCachedSampleRecordId = (modelId: string, recordId: string): void => {
-  if (recordId) {
-    globalThis.localStorage?.setItem(
-      `sampleRecord:${grouparooUiEdition()}:${modelId}`,
-      recordId
+  if (isBrowser() && recordId) {
+    const existing = cookies.get<Record<string, string>>(
```

Comment:
What is `existing` already in the cookie that might want to be preserved?  It's it only this component that writes to that cookie?

Reply:
The solution we went with (paired on this with @edmundito) is to store cached record ids from multiple models in a single cookie via json, keyed by model id.

So a parsed cookie could look like:

```json
{
    "accounts": "abcdf",
    "users": "defgh",
}
```

The library we have for this (universal-cookie), handles reading and writing the whole object but doesn't seem to have a method for updating it in place. Thus we have to `get -> merge -> set` here.

I could rearrange a bit or rename `existing` to make that clearer.